src/peft/tuners/lycoris_utils.py

```diff
@@ -0,0 +1,365 @@
+# coding=utf-8
+# Copyright 2023-present the HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import re
+import warnings
+from dataclasses import dataclass, field
+from itertools import chain
+from typing import Dict, Optional, Set, Type, Union
+
+import torch
+import torch.nn as nn
+from tqdm import tqdm
+
+from peft.config import PeftConfig
+from peft.utils import (
+    ModulesToSaveWrapper,
+    _get_submodules,
+)
+
+from .tuners_utils import BaseTuner, BaseTunerLayer, check_target_module_exists
+
+
+@dataclass
+class LycorisConfig(PeftConfig):
+    r"""
+    A base config for LyCORIS like adapters
+    """
+    rank_pattern: Optional[dict] = field(
```

Comment:
In `LohaConfig` and `LoKrConfig`, the arguments `rank_pattern` and `alpha_pattern` are also defined. Does it make sense to have them in parent class and child class?

Reply:
You are right, I just wanted to leave them explicit in case anyone would read the code, but from the other point it may lead to some errors in case of refactoring or some other modifications, so let's remove them from child classes.